tests/src/FunctionalJavascript/MessageFormTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\FunctionalJavascript;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Component\Utility\Unicode;
+use Drupal\Core\Test\AssertMailTrait;
+
+/**
+ * Test Corporate MessageForm behaviour.
+ */
+class MessageFormTest extends WebDriverTestBase {
+
+  use AssertMailTrait {
+    getMails as drupalGetMails;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'system',
+    'contact',
+    'contact_storage',
+    'oe_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create and login test user.
+    /** @var \Drupal\user\UserInterface $test_user */
+    $testuser = $this->drupalCreateUser([
+      'access site-wide contact form',
+    ]);
+    $this->drupalLogin($testuser);
+  }
+
+  /**
+   * Tests for corporate forms behaviour.
+   */
+  public function testCorporateForm(): void {
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    // Prepare a corporate contact form.
+    $contact_form_id = 'oe_contact_form';
+    $contact_form = ContactForm::create(['id' => $contact_form_id]);
+    $contact_form->setReply('this is a autoreply');
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $topic_label = 'Topic label';
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'topic_label', $topic_label);
+    $email_subject = 'Email Subject';
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'email_subject', $email_subject);
+    $header = 'this is a test header';
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'header', $header);
+    $privacy_text = 'Lorem ipsum sint dolor';
```

Comment:
Let's use "this is a test privacy policy" or something to keep things consistent and readable across the test

Reply:
done